src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+    case 'OneGraphNetlifyCliSessionTestEvent':
+      return friendlyEventName(payload)
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
+      return 'Generate handler as Netlify function '
+    case 'OneGraphNetlifyCliSessionPersistedLibraryUpdatedEvent':
+      return `Sync Netlify Graph operations library`
+    default: {
+      return `Unrecognized event (${__typename})`
+    }
+  }
+}
+
+const handleCliSessionEvent = async ({ authToken, event, netligraphConfig, schema, siteId }) => {
+  const { __typename, payload } = await event
+  switch (__typename) {
+    case 'OneGraphNetlifyCliSessionTestEvent':
+      await handleCliSessionEvent({ authToken, event: payload, netligraphConfig, schema, siteId })
+      break
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
+      await generateHandler(netligraphConfig, schema, payload.operationId, payload)
+      break
+    case 'OneGraphNetlifyCliSessionPersistedLibraryUpdatedEvent':
+      await updateGraphQLOperationsFile({ authToken, docId: payload.docId, netligraphConfig, schema, siteId })
+      break
+    default: {
+      console.warn(
+        `${NETLIFYDEVWARN} Unrecognized event received, you may need to upgrade your CLI version`,
+        __typename,
+        payload,
+      )
+      break
+    }
+  }
+}
+
+const loadCLISession = (state) => state.get('oneGraphSessionId')
+
+const startOneGraphCLISession = async ({ netlifyToken, netligraphConfig, site, state }) => {
+  let oneGraphSessionId = loadCLISession(state)
+  if (!oneGraphSessionId) {
+    const sessionName = generateSessionName()
+    const oneGraphSession = await createCLISession(netlifyToken, site.id, sessionName)
+    state.set('oneGraphSessionId', oneGraphSession.id)
+    oneGraphSessionId = state.get('oneGraphSessionId')
+  }
+
+  const enabledServices = []
+  const schema = await fetchOneGraphSchema(site.id, enabledServices)
+
+  monitorCLISessionEvents({
+    appId: site.id,
+    authToken: netlifyToken,
+    netligraphConfig,
+    sessionId: oneGraphSessionId,
+    state,
+    onEvents: (events) => {
+      events.forEach(async (event) => {
+        const eventName = friendlyEventName(event)
+        log(`${NETLIFYDEVLOG} ${chalk.magenta('Handling')} Netlify Graph event: ${eventName}...`)
+        await handleCliSessionEvent({ authToken: netlifyToken, event, netligraphConfig, schema, siteId: site.id })
+        log(`${NETLIFYDEVLOG} ${chalk.green('Finished handling')} Netlify Graph event: ${eventName}...`)
+      })
+      return events.map((event) => event.id)
+    },
+    onError: (error) => {
+      console.error(`OneGraph: ${error}`)
+    },
+    onClose: () => {
+      console.debug('OneGraph: closed')
+    },
+  })
+}
+
+const fetchAppSchema = async (authToken, siteId) => {
+  const result = await fetchOneGraph(authToken, siteId, internalOperationsDoc, 'AppSchemaQuery', {
+    nfToken: authToken,
+    appId: siteId,
+  })
+
+  return dotProp.get(result, 'data.oneGraph.app.graphQLSchema')
```

Comment:
We don't use this sort of librabry in general, as it makes typos very easy and breaks type inference.

Reply:
Type inference is already broken with these functions, but I'll change it to use nested `&&`